packages/core/admin/admin/src/components/CommandPalette/Groups.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import Items from './Items';
+import Group from './Group';
+
+const Groups = ({ items }) => {
+  const groups = new Map();
+
+  items.forEach((item) => {
+    if (!groups.has(item.group)) {
+      groups.set(item.group, {
+        title: item.group,
+        items: [],
+      });
+    }
+
+    groups.get(item.group).items.push(item);
+  });
```

Comment:
Can you explain what the goal of this is please? Is there a reason we can't do this in the `map` below?

Reply:
we need to group the items par group and then map on groups. I don't see how I could do that without building the data structure 1st. What do you have in mind ?

Reply:
Ah i see. Might be worth adding a small comment as to why we're doing it, it wasn't super clear on first read, I think this is fine though, doesn't need to be overcomplicated.